### PR TITLE
fix(wizard): moved footer to sibling of drawer

### LINKED ISSUE #4890: Wizard - footer updates
**Is this a new component or an extension of an existing one?**
What is the existing component, if any?

**Describe the feature**
A clear and concise description of the new feature. What is the expected behavior?
Move footer outside of drawer and update footer styling/spacing. `pf-c-wizard__footer` should be a sibling of `pf-c-drawer`. 

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
Include screenshots or links to Marvel or other mockups.
Based on @mmenestr 's design/comment in https://github.com/patternfly/patternfly-react/pull/7323#issuecomment-1138753381, we likely want to move the wizard footer out of the drawer to allow for the behavior linked in the design: https://marvelapp.com/prototype/9389af6/screen/87033618 

I fiddled with one of the existing core demos and ended up with this:

Screen Shot 2022-06-02 at 4 41 04 PM

**Any other information?**
Need these core updates done in order to continue work on https://github.com/patternfly/patternfly-react/pull/7323

### LINKED ISSUE #4899: Wizard - "Open drawer" button and styling
**Describe the feature**

Based on @mmenestr 's new design, we need a button to toggle the wizard open and closed. It should be aligned to the top right of the wizard container. 

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**

https://marvelapp.com/prototype/9389af6/screen/86742793

### PR DESCRIPTION
**Pull request by mattnolting**
Closes #4890 
Closes #4899 

https://patternfly-pr-4896.surge.sh/components/wizard#with-drawer
https://patternfly-pr-4896.surge.sh/components/wizard/html-demos#with-drawer
https://patternfly-pr-4896.surge.sh/components/wizard/html-demos#in-page

### CODE REVIEW

**Review by jenny-s51 — APPROVED** [review]
This is looking awesome! 🤩 Thank you @mattnolting !

**Review by mcarrano — APPROVED** [review]
LGTM. Thanks @mattnolting !

**Review by nicolethoen — APPROVED** [review]
Looks good to me!

**Review by mmenestr — APPROVED** [review]

**Comment by mattnolting** [comment]
**Desktop (drawer closed)**

Screen Shot 2022-06-14 at 10 03 50 AM

**Mobile (drawer closed)**

Screen Shot 2022-06-14 at 10 03 34 AM

**In modal (drawer expanded)**

Screen Shot 2022-06-13 at 3 19 32 PM

**Comment by tlabaj** [comment]
Question that pertains to the follow up react work.  Should the toggle always be a link button, or can the consumer use another type of button?

**Comment by mmenestr** [comment]
@tlabaj I would say it should be an icon button or link button yes. As for the tab idea @mattnolting I would just use an icon button - specifically these icons (got them from the sketch symbol library, though I can't find them in the PF icon list): 
image

File: src/patternfly/components/Wizard/examples/Wizard.md [hunk 1]
```diff
@@ -542,8 +611,8 @@ import './Wizard.css'
 ### Accessibility
 | Attribute | Applied to | Outcome |
 | -- | -- | -- |
-| `aria-expanded="true"` | `.pf-c-wizard__toggle` | Indicates that the steps menu is visible. **Required** |
-| `aria-expanded="false"` | `.pf-c-wizard__toggle` | Indicates that the steps menu is hidden. **Required** |
+| `aria-expanded="true"` | `.pf-c-wizard__toggle`, `.pf-c-wizard__drawer-toggle` | Indicates that the steps menu is visible. **Required** |
+| `aria-expanded="false"` | `.pf-c-wizard__toggle`, `.pf-c-wizard__drawer-toggle` | Indicates that the steps menu is hidden. **Required** |
```
Comment (srambach) [inline src/patternfly/components/Wizard/examples/Wizard.md:615]: Is this either the steps menu or drawer is hidden?

**Review by srambach — APPROVED** [review]
Looks good afaict. One wording comment.

File: src/patternfly/components/Wizard/examples/Wizard.md [hunk 2]
```diff
@@ -250,6 +238,87 @@ import './Wizard.css'
...
+    {{/wizard-toggle-list}}
+    {{> wizard-toggle-icon}}
+  {{/wizard-toggle}}
+  {{#> drawer drawer--id="wizard-with-drawer-example-drawer" drawer--IsInline="true"}}
+    {{#> drawer-main}}
+      {{#> drawer-content drawer-content--NoBody="true"}}
+        {{#> wizard-outer-wrap}}
+          {{#> wizard-inner-wrap}}
+            {{#> wizard-main}}
+              {{> wizard-template-nav}}
```
Comment (mcoker) [inline src/patternfly/components/Wizard/examples/Wizard.md:288]: It looks like nav got moved into main in some of these examples - is that intentional? In others it is a child of inner-wrap and comes before main. Seems odd that it's a child of main/main-body.

File: src/patternfly/components/Wizard/wizard.scss [hunk 3]
```diff
@@ -320,6 +329,12 @@
   line-height: var(--pf-c-wizard__toggle-icon--LineHeight);
 }
 
+.pf-c-wizard__drawer-toggle {
```
Comment (mcoker) [inline src/patternfly/components/Wizard/wizard.scss:332]: Could the toggle button get lost if it isn't sticky and is always at the top of the scrolled content?

File: src/patternfly/components/Wizard/wizard.scss [hunk 4]
```diff
@@ -320,6 +329,12 @@
...
+  display: var(--pf-c-wizard__drawer-toggle--Display);
+  float: right;
```
Comment (mcoker) [inline src/patternfly/components/Wizard/wizard.scss:334]: 🥸

Stuff with text is going to flow around it like this - is that intended?

Screen Shot 2022-06-15 at 7 20 42 PM


Screen Shot 2022-06-15 at 7 21 03 PM

Comment (mmenestr) [reply src/patternfly/components/Wizard/wizard.scss:334]: Hmm that's a good point @mcoker - i'd probably at a minimum make sure that the header doesn't wrap around the button like that (or whatever is on the same line as the button). Whatever is below though, like the description, I think is ok to have wrap that way!

**Review by mcoker — CHANGES_REQUESTED** [review]
The demos look good, but looks like we have some double scrollbar action going on, coming from the drawer content element.

Also the nav is cutting off above the footer.

Screen Shot 2022-06-17 at 3 17 56 PM




Here's the wizard contents removed, showing the scrollbar on the drawer content when its empty

Screen Shot 2022-06-17 at 3 16 46 PM

**Review by jenny-s51 — APPROVED** [review]
Approving again for your recent updates - these new examples are looking awesome!! Thank you @mattnolting !

**Comment by mmenestr** [comment]
Everything looks great! Just have a comment about a couple of the demos, which is that when you scroll all the way down in the drawer, I would expect there to be 24px padding at the bottom between the end of the text and the sticky footer, but there isn't - idk if that's because the end of the panel is behind the footer, but I'm wondering if there's a way to change that?

image

https://patternfly-pr-4896.surge.sh/components/wizard/html-demos/with-drawer-expanded/
https://patternfly-pr-4896.surge.sh/components/wizard/html-demos/with-drawer-in-page/

**Comment by mcoker** [comment]
@mmenestr good catch! That's a separate issue with the drawer itself - https://codesandbox.io/s/focused-matsumoto-htkf0m

We have an issue to fix it, but the fix could cause other problems so it's something we're going to address in our a major release - https://github.com/patternfly/patternfly/issues/4797

**Review by mcoker — APPROVED** [review]
LGTM! One question about this demo - https://patternfly-pr-4896.surge.sh/components/wizard/html-demos/with-drawer-in-page/

Should that text say "close drawer" or something to that effect since the drawer is expanded? If so, we can create a follow up.

**Pull request merged**